nav2_ceres_costaware_smoother/include/nav2_ceres_costaware_smoother/ceres_costaware_smoother.hpp

```diff
@@ -0,0 +1,104 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_CERES_COSTAWARE_SMOOTHER__CERES_COSTAWARE_SMOOTHER_HPP_
+#define NAV2_CERES_COSTAWARE_SMOOTHER__CERES_COSTAWARE_SMOOTHER_HPP_
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <algorithm>
+
+#include "nav2_core/smoother.hpp"
+#include "nav2_ceres_costaware_smoother/smoother.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_util/odometry_utils.hpp"
+#include "nav2_util/geometry_utils.hpp"
+#include "geometry_msgs/msg/pose2_d.hpp"
+
+namespace nav2_ceres_costaware_smoother
+{
+
+/**
+ * @class nav2_ceres_costaware_smoother::CeresCostawareSmoother
+ * @brief Regulated pure pursuit controller plugin
+ */
+class CeresCostawareSmoother : public nav2_core::Smoother
+{
+public:
+  /**
+   * @brief Constructor for nav2_ceres_costaware_smoother::CeresCostawareSmoother
+   */
+  CeresCostawareSmoother() = default;
+
+  /**
+   * @brief Destrructor for nav2_ceres_costaware_smoother::CeresCostawareSmoother
+   */
+  ~CeresCostawareSmoother() override = default;
+
+  /**
+   * @brief Configure smoother parameters and member variables
+   * @param parent WeakPtr to node
+   * @param name Name of plugin
+   * @param tf TF buffer
+   * @param costmap_ros Costmap2DROS object of environment
+   */
+  void configure(
+    const rclcpp_lifecycle::LifecycleNode::WeakPtr & parent,
+    std::string name, const std::shared_ptr<tf2_ros::Buffer> & tf,
+    const std::shared_ptr<nav2_costmap_2d::CostmapSubscriber> & costmap_sub,
+    const std::shared_ptr<nav2_costmap_2d::FootprintSubscriber> & footprint_sub) override;
+
+  /**
+   * @brief Cleanup controller state machine
+   */
+  void cleanup() override;
+
+  /**
+   * @brief Activate controller state machine
+   */
+  void activate() override;
+
+  /**
+   * @brief Deactivate controller state machine
+   */
+  void deactivate() override;
+
+  /**
+   * @brief Method to smooth given path
+   *
+   * @param path In-out path to be optimized
+   * @param max_time Maximum duration smoothing should take
+   * @return Smoothed path
+   */
+  bool smooth(
+    nav_msgs::msg::Path & path,
+    const rclcpp::Duration & max_time) override;
+
+protected:
+  std::shared_ptr<tf2_ros::Buffer> tf_;
+  std::string plugin_name_;
+  std::shared_ptr<nav2_costmap_2d::CostmapSubscriber> costmap_sub_;
+  rclcpp::Logger logger_ {rclcpp::get_logger("CeresCostawareSmoother")};
+
+  std::unique_ptr<nav2_ceres_costaware_smoother::Smoother> _smoother;
```

Comment:
Make these 3 last variables have a trailing underscores (e.g. `smoother_`) like the previous 4 class members

Reply:
Actually, you can just make the small number of variable_'s into _variables's, it'll be easier for you than to change all of these all over the place for just 3